docs/proposals/custom-sidecar-config.md

```diff
@@ -0,0 +1,32 @@
+# Custom Container Configuration
+
+## Introduction
+Users and field team have requested access to custom configuration of kubernetes sidecar and init containers. See [#3925](https://github.com/kumahq/kuma/issues/3925). This intuitively aligns with the ProxyTemplate configuration, which provides a free-form configuration of Envoy. The current plan is to introduce a new CRD, similar to ProxyTemplate, which can accommodate sidecar config.
+
+## Custom Resource Definition
+The new CRD will be named ContainerTemplate. It will allow for customer configuration of both sidecar and init containers.
+
+This will follow a similar format to ProxyTemplate. A ContainerTemplate will match pods via namespaces and labels, and modify the containers associated with those pods. The spec will contain full templates for “sidecar” and “init” dataplane containers.
+
+### Example
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: ContainerTemplate
+metadata:
+  name: container-template-1
+spec:
+  selectors:
+    - namespace: namespace-1
```

Comment:
What this `selector` will look like in proto? Predefined fields for `namespace` and pod's name like:
```proto
message Selector {
  string namespace = 1;
  string podName = 2;
}
```
or we want to have it as an arbitrary `map<string, string>`? If the latter, then I guess we should use already defined names like `k8s.kuma.io/namespace` instead of just `namespace`

Reply:
You're right, we don't need proto for this one. But still the question is how we define this `selector`? It can either have `namespace` and `podName` field or it contains only `map[string]string{}` map of tags? It's not clear from the example

Reply:
In the end with switched to annotations so not an issue :)